app/src/main/java/de/westnordost/streetcomplete/overlays/mtb_scale/MtbScaleOverlayForm.kt

```diff
@@ -0,0 +1,109 @@
+package de.westnordost.streetcomplete.overlays.mtb_scale
+
+import android.content.Context
+import android.os.Bundle
+import android.view.View
+import androidx.core.view.isGone
+import com.russhwolf.settings.ObservableSettings
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.UpdateElementTagsAction
+import de.westnordost.streetcomplete.databinding.FragmentOverlayMtbScaleSelectBinding
+import de.westnordost.streetcomplete.overlays.AbstractOverlayForm
+import de.westnordost.streetcomplete.util.LastPickedValuesStore
+import de.westnordost.streetcomplete.util.ktx.valueOfOrNull
+import de.westnordost.streetcomplete.util.logs.Log
+import de.westnordost.streetcomplete.view.setImage
+import org.koin.android.ext.android.inject
+
+class MtbScaleOverlayForm : AbstractOverlayForm() {
+
+    override val contentLayoutResId = R.layout.fragment_overlay_mtb_scale_select
+    private val binding by contentViewBinding(FragmentOverlayMtbScaleSelectBinding::bind)
+
+    private var originalMtbScale: MtbScale? = null
+
+    private lateinit var mtbScaleCtrl: MtbScaleViewController
+    private val prefs: ObservableSettings by inject()
+    private lateinit var favs: LastPickedValuesStore<MtbScale>
+    private val lastPickedMtbScale: MtbScale?
+        get() = favs.get().firstOrNull()
+
+    override fun hasChanges(): Boolean = mtbScaleCtrl.value != originalMtbScale
+
+    override fun isFormComplete(): Boolean = mtbScaleCtrl.value != null
+
+    override fun onClickOk() {
+        val changesBuilder = StringMapChangesBuilder(element!!.tags)
+
+        if (mtbScaleCtrl.value != null) {
+            favs.add(mtbScaleCtrl.value!!)
+        }
+        changesBuilder["mtb:scale"] = mtbScaleCtrl.value!!.osmValue
```

Comment:
sorry for arriving with "everything is more complicated than expected" but we have also `mtb:scale:uphill` https://taginfo.openstreetmap.org/keys/mtb%3Ascale%3Auphill which sadly is quite important

Reply:
yeah there's downhill and uphill. But afair you tag the most difficult grade in mtb:scale and the easier then in downhill/uphill, if one direction is easier. So it isn't a blocker.

Reply:
the worst part is that a second set of images would be best :(

display on map should be fairly doable, maybe with one side for up second for down see sidewalk styling for already existing code

support code for editing both would be  annoying (see surface overlay) - I still vividly remember how annoying it was to code that

Reply:
Well, I don't think that's really required. The [wiki](https://wiki.openstreetmap.org/wiki/Key:mtb:scale#mtb:scale:uphill=0-5) has no set of images for more than a decade. :)

Reply:
> The [wiki](https://wiki.openstreetmap.org/wiki/Key:mtb:scale#mtb:scale:uphill=0-5) has no set of images for more than a decade. :)

well, typically SC has much higher standards than wiki for understandability and clear info (and in some cases wiki has now good images because SC research was applied there)